packages/backend/src/peripherals/database/ProviderCacheRepository.ts

```diff
@@ -1,36 +1,74 @@
 import { Logger } from '@l2beat/backend-tools'
+import { ChainId } from '@lz/libs'
 import type { ProviderCacheRow } from 'knex/types/tables'
 
 import { BaseRepository, CheckConvention } from './shared/BaseRepository'
 import { Database } from './shared/Database'
 
+export interface ProviderCacheRecord {
+  key: string
+  value: string
+  chainId: ChainId
+  blockNumber: number | null
+}
+
 export class ProviderCacheRepository extends BaseRepository {
   constructor(database: Database, logger: Logger) {
     super(database, logger)
     this.autoWrap<CheckConvention<ProviderCacheRepository>>(this)
   }
 
-  async addOrUpdate(row: ProviderCacheRow): Promise<string> {
+  async addOrUpdate(record: ProviderCacheRecord): Promise<string> {
+    const row = toRow(record)
     const knex = await this.knex()
     await knex('provider_cache').insert(row).onConflict('key').merge()
 
     return row.key
   }
 
-  async findByKey(key: string): Promise<ProviderCacheRow | undefined> {
+  async findByKey(key: string): Promise<ProviderCacheRecord | undefined> {
     const knex = await this.knex()
     const row = await knex('provider_cache').where({ key }).first()
-    return row
+
+    if (row) {
+      return toRecord(row)
+    }
   }
 
-  async getAll(): Promise<ProviderCacheRow[]> {
+  async getAll(): Promise<ProviderCacheRecord[]> {
     const knex = await this.knex()
     const rows = await knex('provider_cache').select('*')
-    return rows
+
+    return rows.map(toRecord)
+  }
+  async deleteAfter(blockNumber: number, chainId: ChainId): Promise<number> {
```

Comment:
```suggestion
  }
  
  async deleteAfter(blockNumber: number, chainId: ChainId): Promise<number> {
```

Reply:
And please write a test for `deleteAfter`

Reply:
Here ya go
https://github.com/l2beat/lz-monitoring/pull/37/commits/2e7ea717abeaeb3373835ccca23bd0573ef2206b